azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -86,12 +107,84 @@ public ProjectManager(ProjectLoader loader, Props props) {
     new XmlValidatorManager(prop);
     loadAllProjects();
     loadProjectWhiteList();
+
+    CleanerThread cleanerThread = new CleanerThread();
+    cleanerThread.start();
+  }
+
+  /**
+   * Cleaner thread to clean projects. Runs every day.
+  **/
+  private class CleanerThread extends Thread {
+
+    // runs every day
+    private static final long CLEANER_THREAD_WAIT_INTERVAL_MS = 24 * 60 * 60 * 1000;
+    private final User deleter = new User("azkaban");
+
+    private boolean shutdown = false;
+    private long lastProjectFileCleanTime = -1;
+
+    public CleanerThread() {
+      this.setName("ProjectManager-Cleaner-Thread");
+      logger.info("Starting cleaner thread: " + this.getName());
+    }
+
+    @SuppressWarnings("unused")
+    public void shutdown() {
+      shutdown = true;
+      this.interrupt();
+    }
+
+    public void run() {
+      while (!shutdown) {
```

Comment:
Can we add the reason for for removing Synchronized as a  comment

Reply:
Done